minio/api.py

```diff
@@ -619,7 +619,7 @@ def make_bucket(self, bucket_name, location=None, object_lock=False):
                         self._base_url.region, location,
                     ),
                 )
-        location = location or "us-east-1"
+        location = self._base_url.region or location or "us-east-1"
```

Comment:
```suggestion
        location = location or self._base_url.region or "us-east-1"
```

Reply:
Should it be this, as the user is explicitly specifying the location they want to use, even if the region is set in the constructor?

Reply:
No. It is not allowed. It will make confusion in other APIs.

Reply:
This is a general comment.
Wouldn't it be better to define a constant for the default region instead of hard coding it in several different places (5) in the code?
```
_DEFAULT_REGION = "us-east-1"
```

Reply:
I think this is out of scope of this PR.

Reply:
so, if `region` is specified in the constructor, and also specified in the `make_bucket` call and they are different, do we reject it?

Reply:
yep